src/audio/dai.c

```diff
@@ -728,10 +757,30 @@ static int dai_config(struct comp_dev *dev, struct sof_ipc_dai_config *config)
 			dd->chan = DMA_CHAN_INVALID;
 		}
 		break;
+	case SOF_DAI_INTEL_ALH:
+		/* set to some non-zero value to satisfy the condition below,
+		 * it is recalculated in dai_params() later
+		 */
+		dd->frame_bytes = 4;
+
+		/* As with HDA, the DMA channel is assigned in runtime,
+		 * not during topology parsing.
+		 */
+		channel = config->alh.stream_id;
+		dd->stream_id = config->alh.stream_id;
+		trace_dai_with_ids(dev, "dai_config(), channel = %d",
+				   channel);
+
+		if (dd->chan != DMA_CHAN_INVALID) {
+			dma_channel_put(dd->dma, dd->chan);
+			dd->chan = DMA_CHAN_INVALID;
+		}
```

Comment:
hi @slawblauciak  please delete the following line
````
		if (dd->chan != DMA_CHAN_INVALID) {
			dma_channel_put(dd->dma, dd->chan);
			dd->chan = DMA_CHAN_INVALID;
		}
````
Now sdw doesn't work with pulse audio and the reason is that: dai_config will be triggered for both playback and
capture to free dma channel for they use the same dai_index(please check https://github.com/thesofproject/sof/blob/master/src/ipc/ipc.c#L406). For HDA, playback and capture use different dai_index. So if both playback & capture are working, and playback is stopped then capture failed.

Alh uses dw-dma, not link-dma. For link-dma, the dma channel is specialized by kernel driver,
but there is no such requirement for dw-dma. Actually ahl is closer to I2S than HDA. please check there is no
such operation for I2S(SSP).